core/consensus/babe/impl/babe_impl.cpp

```diff
@@ -116,22 +118,44 @@ namespace kagome::consensus {
     runSlot();
   }
 
-  void BabeImpl::processSlotLeadership(const crypto::VRFOutput &output) {
-    // build a block to be announced
-    auto &&[_, best_block_hash] = block_tree_->deepestLeaf();
-
+  outcome::result<primitives::PreRuntime> BabeImpl::babePreDigest(
```

Comment:
The name of the method is not quite clear to me. Does it produce a pre-digest?

Reply:
yes! PreRuntime is one of possible digest items. Actually, this name was copied from substrate